src/lints/trait_method_added.ron

```diff
@@ -0,0 +1,61 @@
+SemverQuery(
+    id: "trait_method_added",
+    human_readable_name: "pub trait method added",
+    description: "A non-sealed public trait added a new method without a default implementation",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#trait-new-item-no-default"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        sealed @filter(op: "!=", value: ["$true"])
+                        name @output
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        method {
+                            method_name: name @output @tag
+                            has_body @filter(op: "!=", value: ["$true"])
+
+                            span_: span @optional {
+                                filename @output
+                                begin_line @output
+                            }
+                        }
```

Comment:
This portion is correctly implemented, but would be lovely if we had a test case for one more situation that users might find confusing.

Users might think that marking the new method `#[doc(hidden)]` may resolve this lint. It shouldn't, and it'd be great to have a test for it.

For bonus points, if you're interested, you could take a look at our other "trait item added without default" lints and make sure they all have a "`#[doc(hidden)]` on the new item" test case as well. If not, more PRs are welcome!